// watson-react-components/dist/components/Header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Header;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Sizes = require('./Sizes');

var _Sizes2 = _interopRequireDefault(_Sizes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint max-len:off */
function Header(prop) {
  return _react2.default.createElement(
    'header',
    { className: 'header' },
    _react2.default.createElement(
      'div',
      {
        className: prop.size === 'medium' ? '_container' : '_container _container_' + prop.size
      },
      _react2.default.createElement(
        'div',
        { className: 'header--breadcrumbs' },
        prop.mainBreadcrumbs ? _react2.default.createElement(
          'a',
          {
            href: prop.mainBreadcrumbsUrl,
            className: 'base--a header--breadcrumbs-link header--breadcrumbs-main'
          },
          prop.mainBreadcrumbs
        ) : null,
        prop.subBreadcrumbs ? _react2.default.createElement(
          'div',
          { style: { marginTop: '0rem', display: 'inline-block' } },
          _react2.default.createElement(
            'span',
            { className: 'header--breadcrumbs-slash' },
            ' / '
          ),
          _react2.default.createElement(
            'a',
            {
              href: prop.subBreadcrumbsUrl,
              className: 'base--a header--breadcrumbs-link header--breadcrumbs-sub'
            },
            prop.subBreadcrumbs
          )
        ) : null
      )
    )
  );
}

Header.propTypes = {
  mainBreadcrumbs: _react2.default.PropTypes.string,
  mainBreadcrumbsUrl: _react2.default.PropTypes.string,
  subBreadcrumbs: _react2.default.PropTypes.string,
  subBreadcrumbsUrl: _react2.default.PropTypes.string,
  hasWordmark: _react2.default.PropTypes.bool,
  size: _react2.default.PropTypes.oneOf(_Sizes2.default)
};

Header.defaultProps = {
  hasWordmark: true,
  size: 'large'
};
module.exports = exports['default'];
