// watson-react-components/dist/components/Jumbotron.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Jumbotron;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Icon = require('./Icon');

var _Sizes = require('./Sizes');

var _Sizes2 = _interopRequireDefault(_Sizes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Jumbotron(props) {
  return _react2.default.createElement(
    'section',
    { className: '_full-width-row _full-width-row_gray jumbotron' },
    _react2.default.createElement(
      'div',
      {
        className: props.size === 'medium' ? '_container' : '_container _container_' + props.size
      },
      _react2.default.createElement(
        'div',
        { className: 'jumbotron--heading-section' },
        _react2.default.createElement(
          'h2',
          { className: 'base--h2 jumbotron--header' },
          props.serviceName
        ),
        _react2.default.createElement(
          'p',
          { className: 'base--p jumbotron--description description' },
          props.description
        )
      ),
      props.version !== 'GA' ? _react2.default.createElement(
        'div',
        { className: 'jumbotron--version-tag' },
        props.version
      ) : null
    )
  );
}

Jumbotron.propTypes = {
  documentation: _react2.default.PropTypes.string,
  version: _react2.default.PropTypes.string,
  description: _react2.default.PropTypes.string.isRequired,
  serviceName: _react2.default.PropTypes.string.isRequired,
  apiReference: _react2.default.PropTypes.string,
  startInBluemix: _react2.default.PropTypes.string,
  repository: _react2.default.PropTypes.string,
  size: _react2.default.PropTypes.oneOf(_Sizes2.default)
};

Jumbotron.defaultProps = {
  size: 'large'
};
module.exports = exports['default'];
